doc/source/getting_started/code/ODL Basics.py

```diff
@@ -0,0 +1,198 @@
+
+# Copyright 2014-2016 The ODL development group
+#
+# This file is part of ODL.
+#
+# ODL is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# ODL is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with ODL.  If not, see <http://www.gnu.org/licenses/>.
+
+
+# # ODL Basics
+
+# Welcome to your first tutorial. Here you will learn the basics of linear
+# spaces, vectors and related functionality as used throughout ODL. The
+# main purpose is to get you accustomed to working with the library, the
+# "look and feel" of typical code and some building blocks and patterns
+# that repeat themselves in other parts. We will explain them as we go
+# along.
+#
+# At this point, you should have a working installation of ODL on your
+# machine. If that's not the case, check out [the installation instruction
+# s](https://odlgroup.github.io/odl/getting_started/installing.html) and
+# come back when you're finished!
+
+# We import this for Python 2 compatibility
+from __future__ import print_function
+import odl
+
+
+# Everything starts with importing the package. If you want to explore it,
+# you can type `odl.` and hit `<TAB>` in an interactive shell (for example
+# [IPython](https://ipython.org/)) to see all its top-level contents. The
+# `odl` package is organized into several sub-packages, but the most
+# important core functions and classes are available at the top level, so
+# you don't have to type `odl.subpackage1.subpackage2.module.function` all
+# the time.
+#
+# Now the simplest kind of mathematical object you can create in ODL is a
+# *vector*. If you're familiar with [NumPy](http://www.numpy.org/), you
+# can simply think of a `numpy.ndarray`.
+
+x = odl.vector([1, 2, 3])
+print(x)
+
+
+# In fact, the default implementation of a `vector` is just a thin wrapper
+# around `numpy.ndarray`. However, this back-end can be switched for
+# something else, e.g., a CUDA-based data container. On the surface it
+# will look exactly the same, and the differences will be entirely under
+# the hood. More on that later.
+#
+# Let's look at the representation (`repr`) of `x` to get some additional
+# information beyond the pretty-printing above:
+
+x  # or print(repr(x))
+
+
+# Hm, this is interesting. Obviously, there's more to this vector than
+# just its values. The line printed above suggests that it is an `element`
+# of something called `fn`. Let's see if we can just look at the first
+# part of that expression, of course prepending the package name and a
+# dot, a trick we know from NumPy:
+
+odl.fn(3, 'int')
```

Comment:
dtypes should *not* come in the first example, change to floats and pretend it doesn't happen.